src/fidesops/service/privacy_request/request_runner_service.py

```diff
@@ -25,79 +29,89 @@
 from fidesops.util.async_util import run_async
 from fidesops.util.cache import FidesopsRedis
 
+logger = logging.getLogger(__name__)
+
+_scheduler = None
+
+
+def get_scheduler() -> BackgroundScheduler:
+    """Returns a BackgroundScheduler as a singleton"""
+    global _scheduler  # pylint: disable=W0603
+    if _scheduler is None:
+        _scheduler = BackgroundScheduler()
+        _scheduler.start()
+    return _scheduler
```

Comment:
Is instantiating in this way vulnerable to a race condition?

Reply:
This is a broken singleton pattern, e.g. https://stackoverflow.com/a/33000332/19479 .We're instantiating on demand here. Granted the worst that could happen is an additional scheduler can be created.

Reply:
ah thank you for clarifying! this link is super helpful.

Reply:
@stevenbenjamin when you get a chance, i've tried to fix this in the latest commit -